File: short_problems/C++/alternating_characters.cpp
#include <iostream>
#include <string>
using namespace std;

int altern_chars(const string &s) {
    int count = 0;
    for (int i = 1; i < s.size(); i++) {
        if (s.at(i) == s.at(i - 1)) {
            count++;
        }
    }
    return count;
}

int main() {
    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        string s;
        cin >> s;
        cout << altern_chars(s) << endl;
    }
    return 0;
}

File: short_problems/Python/alternating_characters.py
#!/usr/local/bin/python3

def altern_chars(s):
    count = 0
    for i in range(1, len(s)):
        if s[i] == s[i - 1]:
            count += 1
    return count

if __name__ == "__main__":
    num_cases = int(input())
    for i in range(num_cases):
        s = str(input())
        print(altern_chars(s))

